siteimprove/admin/js/siteimprove.js

```diff
@@ -51,7 +72,44 @@
         return;
       }
 
-      _si.push([this.method, this.url, this.token]);      
+      _si.push(['onHighlight', function(highlightInfo) {
+        // Remove highlight tag wrapper
+        $(".si-highlight").contents().unwrap();
+        // Create an span tag for every highlight
+        $.each(highlightInfo.highlights, function(index, highlight) {
+          var $element = $(highlight.selector);
+          var text = $element.text();
+          
+          if (highlight.offset) {
+            var start = highlight.offset.start;
+            var length = highlight.offset.length;
+            
+            var before = text.substr(0, start);
+            var highlighted = text.substr(start, length);
+            var after = text.substr(start + length);
+            
+            $element.html(before + "<span class='si-highlight'>" + highlighted + "</span>" + after);
+          } else {
+            $element.html("<span class='si-highlight'>" + text + "</span>");
+          }
+        });
+      }]);
+
+      //Adaptation carried out to comply with the CMS-plugin-v2 documentation
+      if( this.method === "domain" ){
```

Comment:
With the newest changes to overlay.js we should on longer fail if the "domain" function is called - it should just ignore it. So I think you could remove this "fix" for now - to not break v1 functionality.